riscv/insns/mnret.h

```diff
@@ -0,0 +1,15 @@
+require_extension(EXT_SMRNMI);
+require_privilege(PRV_M);
+set_pc_and_serialize(p->get_state()->mnepc->read());
+reg_t s = STATE.mnstatus->read();
+reg_t prev_prv = get_field(s, MNSTATUS_MNPP);
+reg_t prev_virt = get_field(s, MNSTATUS_MNPV);
+if (prev_prv != PRV_M) {
+  reg_t mstatus = STATE.mstatus->read();
+  mstatus = set_field(mstatus, MSTATUS_MPRV, 0);
+  p->put_csr(CSR_MSTATUS, mstatus);
```

Comment:
Why not `STATE.mstatus->write(mstatus)`? That would avoid an unnecessary map lookup in `csrmap`. (Same comment for the other `put_csr()` just below.)

Could even add a simple method to `mstatus_csr_t` so this could do `STATE.mstatus->clear_mprv()`.

Reply:
Copy-pasta from the implementation of the MRET instruction.

I decided against defining `clear_mprv`, since inlining the code isn't so bad and I don't expect the method to get much reuse.